runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/FnApiControlClientPoolService.java

```diff
@@ -79,7 +76,7 @@ public static FnApiControlClientPoolService offeringClientsToPool(
       // discarded, which should be performed by a call to #shutdownNow. The remote caller must be
       // able to handle an unexpectedly terminated connection.
       vendedClients.add(newClient);
```

Comment:
<!--thread_id:cc_182896326_t; commit:1fd121da1417624b3b84f0300648251da64b9cb5; resolved:0-->
<!--section:context-quote-->
> **jkff** wrote:
> Is this thread-safe?

<!--section:body-->
`grpc-java` is _definitely_ thread-safe within callbacks on a given `StreamObserver` due to requests being handled sequentially. What I don't know is whether incoming client connections are serialized. @tgroh Do you know the answer to this?